WeatherApp.tsx

```diff
@@ -0,0 +1,81 @@
+import React, { useEffect, useState } from 'react';
+import { View, StyleSheet, ActivityIndicator } from 'react-native';
+import { useNavigation } from '@react-navigation/native';
+import { createDrawerNavigator } from '@react-navigation/drawer';
+import * as Location from 'expo-location';
+import useWeatherAPI from './useWeatherAPI';
+import CurrentWeatherScreen from './CurrentWeatherScreen';
+import ForecastScreen from './ForecastScreen';
+
+const Drawer = createDrawerNavigator();
+const API_KEY = 'd7e88c2243f84a3eb41183055231408';
+
+const WeatherApp = () => {
+  const navigation = useNavigation();
+  const [location, setLocation] = useState(null);
+
+  useEffect(() => {
+    (async () => {
+      try {
+        const { status } = await Location.requestForegroundPermissionsAsync();
+        if (status === 'granted') {
+          const location = await Location.getCurrentPositionAsync({});
+          setLocation({
+            latitude: location.coords.latitude,
+            longitude: location.coords.longitude,
+          });
+        }
+      } catch (error) {
+        console.error('Error getting location:', error);
+      }
+    })();
+  }, []);
+
+  const { data: currentWeatherData, loading: currentWeatherLoading } = useWeatherAPI(
```

Comment:
Would have liked to see some mapping functions like I had shared in the resources section. This way we can use typescript to make our response type-safe and we can continue the camelCase convention instead of snake_case like the API provides.